### PR TITLE
feat: Support the SparkFun ProMini 3V

### PR DESCRIPTION
**Pull request by emilbonnek**
This PR adds a new feature to arduino-hal, `sparkfun-promini-3v`, and a new board to ravedude, `promini-3v`. It is a copy, in nearly all respects, of the 5v implementation. The only difference is the clock speed of 8 MhZ.

EDIT: I just learned that the 3v board is actually 3.3v so maybe it should be named `promini-3v3` instead for clarification?

### CODE REVIEW

**Comment by stappersg** [comment]
On Mon, Nov 06, 2023 at 04:45:34AM -0800, Emil Bonne Kristiansen wrote:
> The only difference is the clock speed of 8 MhZ.

That brings back a memory of a simular request.

Someone had a board with an AVR 328, copied Arduino board,
modified only clock speed to match the boards clock speed.

I don't know by heart how it was handled.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by emilbonnek** [comment]
@Rahix would you mind taking a look at this at your leisure?

**Comment by stappersg** [comment]
> That brings back a memory of a simular request.
> 
> Someone had a board with an AVR 328, copied Arduino board, modified only clock speed to match the boards clock speed.
>
> I don't know by heart how it was handled.

It was _"to be easier to treat it as an arduino (with an 8Mhz clock speed)"_ from https://github.com/Rahix/avr-device/issues/124#issuecomment-1507467463

Now I know this merge request is about something different.

**Comment by Rahix** [comment]
@emilbonnek thanks a lot for the contribution and sorry for loosing track of this.  I'm now in the situation where a second PR exists that also implements pretty much exactly the same feature...

I've chosen to merge #491 in favor of this one, simply because that one already had the merge-conflicts resolved.  Hope you understand that your contribution is appreciated nonetheless, this is just a result of me loosing track :(
